raiden-ts/src/utils/ethers.ts

```diff
@@ -43,29 +53,44 @@ export function fromEthersEvent<T>(
       resultSelector,
     ) as Observable<T>;
 
-  let first = 0;
-  let latestEventBlock = 0;
+  const blockQueue: number[] = []; // sorted 'fromBlock' queue, at most of [range] size
   return defer(() => {
-    // 'first' starts with subscription-time's private value set with provider.resetEventsBlock
-    first = (target as any)._lastBlockNumber || target.blockNumber || Number.POSITIVE_INFINITY;
+    // 'resetEventsBlock' is private, set at [[Raiden]] constructor, so we need 'any'
+    const resetBlock: number = (target as any)._lastBlockNumber;
+    const firstBlock = resetBlock && resetBlock > 0 ? resetBlock : target.blockNumber ?? 1;
+    blockQueue.push(firstBlock); // starts 'blockQueue' with subscription-time's resetEventsBlock
```

Comment:
What is the purpose of first initiating an empty array and then pushing its first entry. Why not making it simple and say `const blockQueue = [firstBlock]`? :thinking:

Reply:
the `blockQueue.push` is called inside the `defer`, i.e. at **subscription** time, while the queue is defined outside of this context so it's available also for the following functions.